### PR TITLE
Document Keyboard class and add logging support

### PR DESCRIPTION
**Pull request by frostburn**

### CODE REVIEW

File: src/keyboard.ts [hunk 1]
```diff
@@ -154,8 +178,18 @@ export class Keyboard {
       window.addEventListener("keydown", this._keydown);
       window.addEventListener("keyup", this._keyup);
     }
+
+    if (log === undefined) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
```
[the remark this answers is not in the document]
Comment (frostburn) [reply src/keyboard.ts:183]: That's not what this directive is about. It refers to the `msg` variable.

File: src/keyboard.ts [hunk 2]
```diff
@@ -165,48 +199,70 @@ export class Keyboard {
...
     this.keydownCallbacks.forEach((callback) =>
       keyupCallbacks.push(callback(event))
     );
     this.keyupCallbacks.set(event.code, keyupCallbacks);
   }
 
-  fireKeyup(event: CoordinateKeyboardEvent) {
+  private fireKeyup(event: CoordinateKeyboardEvent) {
+    this.log(
+      `Firing keyup listeners with ${event.code} @ ${event.coordinates}`
```
[the remark this answers is not in the document]
Comment (frostburn) [reply src/keyboard.ts]: Good point. I removed that part.

**Comment by frostburn** [comment]
Checks pass. I consider the eslint directive meta-issue to be minor so I'll go ahead and merge this.

File: src/keyboard.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (forrcaho) [reply src/keyboard.ts:183]: The code you have is fine, but just to be clear on what I meant here
```ts
  constructor(autobind = false, log?: (msg: string) => void = (msg) => {}) {
  this.log = log;
```
wouldn't require an eslint directive.

[the remark this answers is not in the document]
Comment (frostburn) [reply src/keyboard.ts:183]: Did you check that it doesn't? I find it odd that eslint wouldn't complain about an unused `msg` variable and an empty function body here as well. (We have the empty body checks disabled project wide.)

[the remark this answers is not in the document]
Comment (forrcaho) [reply src/keyboard.ts:183]: Yes, I tried this version and `npm run lint` came up clean.

**Pull request merged**